apps/hash-graph/libs/api/src/rpc/account.rs

```diff
@@ -0,0 +1,704 @@
+                    .map_body(Either::Left)
+                    .map_body(Either::Right)
+                    .map_body(Either::Right)
+            }),
+            meta::AccountProcedureId::AddAccountGroupMember => delegate_call_discrete(
+                request,
+                codec,
+                |(account_group_id, account_id)| async move {
+                    self.inner
+                        .add_account_group_member(session, account_group_id, account_id)
+                        .await
+                },
+            )
+            .await
+            .map(|response| {
+                response
+                    .map_body(Either::Left)
+                    .map_body(Either::Right)
+                    .map_body(Either::Right)
+                    .map_body(Either::Right)
+            }),
+            meta::AccountProcedureId::RemoveAccountGroupMember => delegate_call_discrete(
+                request,
+                codec,
+                |(account_group_id, account_id)| async move {
+                    self.inner
+                        .remove_account_group_member(session, account_group_id, account_id)
+                        .await
+                },
+            )
+            .await
+            .map(|response| {
+                response
+                    .map_body(Either::Right)
+                    .map_body(Either::Right)
+                    .map_body(Either::Right)
+                    .map_body(Either::Right)
+            }),
+        }
+    }
+}
+
+// TODO: this can be auto generated by the `harpc` crate
+#[derive(Debug, Copy, Clone, PartialEq, Eq, Hash)]
+pub struct AccountClient;
+
+impl<Svc, C> AccountService<role::Client<Svc, C>> for AccountClient
+where
+    Svc: harpc_client::connection::ConnectionService<C>,
+    C: harpc_client::connection::ConnectionCodec,
+{
+    async fn create_account(
+        &self,
+        session: Connection<Svc, C>,
+        params: InsertAccountIdParams,
+    ) -> Result<AccountId, Report<AccountError>> {
+        invoke_call_discrete(session, meta::AccountProcedureId::CreateAccount, [params])
+            .await
+            .change_context(AccountError)
+    }
+
+    async fn create_account_group(
+        &self,
+        session: Connection<Svc, C>,
+        params: InsertAccountGroupIdParams,
+    ) -> Result<AccountGroupId, Report<AccountError>> {
+        invoke_call_discrete(session, meta::AccountProcedureId::CreateAccountGroup, [
+            params,
+        ])
+        .await
+        .change_context(AccountError)
+    }
+
+    async fn check_account_group_permission(
+        &self,
+        session: Connection<Svc, C>,
+        account_group_id: AccountGroupId,
+        permission: AccountGroupPermission,
+    ) -> Result<PermissionResponse, Report<AccountError>> {
+        invoke_call_discrete(
+            session,
+            meta::AccountProcedureId::CheckAccountGroupPermission,
+            [(account_group_id, permission)],
```

Comment:
We have discussed this offline, but to recap:

procedure calls operate on streams that are input and then output again. This allows us to _defer_ information in some cases or even embed control flow in the future using e.g. `async gen` syntax.

This is why I introduced in this PR `invoke_call_discrete`, instead of being a continuous stream, it instead takes an iterator, then creates a stream of items out of them that are given to the server. In our case it's only a single item, but in theory could be multiple items. Take for example:

```
pub trait AccountService<R> {
	pub fn check_account_group_permission(&self, pairs: Vec<(AccountGroupId, Permission)>);
}
```

> This is a very contrived example, something you wouldn't really expect to be a real usecase.

We could in theory send these items all at the same time, or one by one over the wire. A single `Iterator` corresponds always to a single call.